addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py

```diff
@@ -895,13 +909,15 @@ def _correct_invoice_tax_amount(self, tree, invoice):
                 taxes = invoice.line_ids.tax_line_id.filtered(lambda tax: tax.amount == tax_percent)
                 # If we found taxes with the correct amount, look for a tax line using it, and correct it as needed.
                 if taxes:
-                    tax_total = float(amount.text)
-                    tax_line = invoice.line_ids.filtered(lambda line: line.tax_line_id in taxes)[:1]
-                    if tax_line:
+                    tax_total = document_amount_sign * float(amount.text)
```

Comment:
Now that I understand the `qty_factor` better, maybe we could just add it as kwarg? (and remove `document_amount_sign`)
```suggestion
                    tax_total = qty_factor * float(amount.text)
```